pkg/services/accesscontrol/evaluator.go

```diff
@@ -80,6 +84,25 @@ func match(scope, target string) bool {
 	return scope == target
 }
 
+func (p permissionEvaluator) EvaluateCustom(fn CheckerFn) (bool, error) {
```

Comment:
We can probably replace `Evaluate` with this and run extract the match function but lets leave it for later